src/main/java/org/launhcode/healthynutrition/controllers/ProfileController.java

```diff
@@ -0,0 +1,89 @@
+package org.launhcode.healthynutrition.controllers;
+
+
+import org.launhcode.healthynutrition.data.UserRepository;
+import org.launhcode.healthynutrition.models.Profile;
+import org.launhcode.healthynutrition.models.User;
+import org.launhcode.healthynutrition.models.data.ProfileRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.Optional;
+
+@Controller
+@RequestMapping("/profile")
+public class ProfileController {
+
+    @Autowired
+    private ProfileRepository repository;
+
+    @Autowired
+    private UserRepository userRepo;
+
+
+    @GetMapping
+    public String profileForm(Model model) {
+        model.addAttribute("profile", new Profile());
+        return "profile";
+    }
+
+    @PostMapping
+    public String createProfile(Model model, Profile profile) {
+        ;
+        model.addAttribute("profile");
+        repository.save(profile);
+        return "redirect:";
+    }
+
+    @GetMapping("/{id}")
+    public String displayEditProfileForm(Model model, @PathVariable("id") int id) {
+
+        Optional<Profile> optional = repository.findById(id);
+        Profile profile =  null;
+        if (optional.isEmpty()) {
+            profile = new Profile();
```

Comment:
Is this for when the user is editing the profile for the first time?

Reply:
when we look up with id, if there is no id with existing user then will create new profile for that.